diff --git a/.github/workflows/cross.yml b/.github/workflows/cross.yml
index f75102a343fd4bcf07e753075686597c9b42d858..04ebf86fe7ec96276514c8085b6e1ea1fb66fd22 100644
--- a/.github/workflows/cross.yml
+++ b/.github/workflows/cross.yml
@@ -5,7 +5,7 @@ concurrency:
   cancel-in-progress: true
 jobs:
   build:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
     name: '${{ matrix.target.arch }}, ${{ matrix.sys.compiler }} ${{ matrix.sys.version }}'
     strategy:
       matrix:
@@ -14,7 +14,7 @@ jobs:
           - { platform: 'arm',     arch: 'armv7-a', dir: 'arm-linux-gnueabihf', flags: '-mfpu=vfpv3-d16', full: 'OFF' } # no neon
           - { platform: 'aarch64', arch: 'armv8-a', dir: 'aarch64-linux-gnu', flags: '', full: 'ON' }
         sys:
-          - { compiler: 'gcc',   version: '8' }
+          - { compiler: 'gcc',   version: '9' }
           - { compiler: 'clang', version: 'latest' }
     steps:
     - name: Setup compiler
diff --git a/include/xsimd/arch/generic/xsimd_generic_details.hpp b/include/xsimd/arch/generic/xsimd_generic_details.hpp
index e676e0a7d23036298ae4ca0b16d75eb441579811..a9af608c88c5616a41b8af7da448c3bbce8baebf 100644
--- a/include/xsimd/arch/generic/xsimd_generic_details.hpp
+++ b/include/xsimd/arch/generic/xsimd_generic_details.hpp
@@ -204,11 +204,11 @@ namespace xsimd
             XSIMD_INLINE batch<uint32_t, A> fast_cast(batch<float, A> const& v, batch<uint32_t, A> const&, requires_arch<generic>) noexcept
             {
                 auto is_large = v >= batch<float, A>(1u << 31);
-                auto small = bitwise_cast<float>(batch_cast<int32_t>(v));
-                auto large = bitwise_cast<float>(
+                auto small_v = bitwise_cast<float>(batch_cast<int32_t>(v));
+                auto large_v = bitwise_cast<float>(
                     batch_cast<int32_t>(v - batch<float, A>(1u << 31))
                     ^ batch<int32_t, A>(1u << 31));
-                return bitwise_cast<uint32_t>(select(is_large, large, small));
+                return bitwise_cast<uint32_t>(select(is_large, large_v, small_v));
             }
         }
 
