diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2006604..4f4f4d1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -323,31 +323,33 @@ if (USE_THREAD)
   endif()
 endif()
 
-#if (MSVC OR NOT NOFORTRAN)
-if (NOT NO_CBLAS)
-  if (NOT ONLY_CBLAS)
-  # Broken without fortran on unix
-    add_subdirectory(utest)
-endif()
-endif()
-
-if (NOT NOFORTRAN)
- if (NOT ONLY_CBLAS)
-  # Build test and ctest
-  add_subdirectory(test)
- endif()
- if (BUILD_TESTING AND NOT BUILD_WITHOUT_LAPACK)
-    add_subdirectory(lapack-netlib/TESTING)
+if (BUILD_TESTING)
+  #if (MSVC OR NOT NOFORTRAN)
+  if (NOT NO_CBLAS)
+    if (NOT ONLY_CBLAS)
+    # Broken without fortran on unix
+      add_subdirectory(utest)
+    endif()
   endif()
-endif()
-  if(NOT NO_CBLAS)
+
+  if (NOT NOFORTRAN)
    if (NOT ONLY_CBLAS)
-    add_subdirectory(ctest)
+    # Build test and ctest
+    add_subdirectory(test)
    endif()
+   if (BUILD_TESTING AND NOT BUILD_WITHOUT_LAPACK)
+      add_subdirectory(lapack-netlib/TESTING)
+    endif()
+  endif()
+  if(NOT NO_CBLAS)
+    if (NOT ONLY_CBLAS)
+      add_subdirectory(ctest)
+    endif()
   endif()
   if (CPP_THREAD_SAFETY_TEST OR CPP_THREAD_SAFETY_GEMV)
     add_subdirectory(cpp_thread_test)
   endif()
+endif()
 
 if (NOT FIXED_LIBNAME)
 set_target_properties(${OpenBLAS_LIBS} PROPERTIES
